packages/pieces/framework/src/lib/property/dropdown-prop.ts

```diff
@@ -5,29 +5,35 @@ import { OAuth2PropertyValue } from "./oauth2-prop";
 
 export type DropdownState<T> = {
 	disabled?: boolean;
-	placeholder?: string;
-	options: DropdownOption<T>[];
+
 }
 
 export type DropdownOption<T> = {
 	label: string;
 	value: T;
 };
 
+// MainDropdownState extends DropdownState and adds placeholder and options
+export type MainDropdownState<T> = DropdownState<T> & {
+	placeholder?: string;
+	options: DropdownOption<T>[];
+  };
+
+
 export type DropdownProperty<T, R extends boolean> = BasePropertySchema & {
 	refreshers: string[];
-	options: (propsValue: Record<string, undefined | OAuth2PropertyValue | number | string | object | BasicAuthPropertyValue | unknown[]>) => Promise<DropdownState<T>>
+	options: (propsValue: Record<string, undefined | OAuth2PropertyValue | number | string | object | BasicAuthPropertyValue | unknown[]>) => Promise<MainDropdownState<T>>
 } & TPropertyValue<T, PropertyType.DROPDOWN, R>;
 
 export type StaticDropdownProperty<T, R extends boolean> = BasePropertySchema & {
-	options: DropdownState<T>;
+	options: MainDropdownState<T>;
```

Comment:
This still has the problem in the issue, where the user have to type
options inside options in static dropdown.

I am not sure why did you create MainDropdownState in that case

Reply:
Turns out I didn't look at the problem that way, MainDropdownState was supposed to make more sense but ended up replicating the problem, however I've come up with a better solution.